CI: ci-clustermesh: check-log-errors: Failed while waiting for initial global identities

## CI failure

Hit on https://github.com/cilium/cilium/pull/40863 which has a small metrics bugfix and a new linter, so the failure should be unrelated to those changes.

https://github.com/cilium/cilium/actions/runs/16729347210/job/47353347967#step:53:480

```
📋 Test Report [cilium-test-1]
❌ 1/8 tests failed (1/168 actions), 117 tests skipped, 3 scenarios skipped:
Test [check-log-errors]:
  🟥 check-log-errors/no-errors-in-logs:pkg/endpoint:kind-cluster2/kube-system/cilium-6rwbw (cilium-agent): Found 1 logs in kind-cluster2/kube-system/cilium-6rwbw (cilium-agent) matching list of errors that must be investigated:
time=2025-08-04T17:08:47.700575884Z level=warn source=/go/src/github.com/cilium/cilium/pkg/endpoint/restore.go:308 msg="Failed while waiting for initial global identities" ciliumEndpointName=kube-system/clustermesh-apiserver-5777d847d9-j4vh7 endpointID=192 containerInterface=eth0 identity=16728613 datapathPolicyRevision=0 ipv4=10.244.2.190 desiredPolicyRevision=0 containerID=d698eee51b ipv6=fd00:10:244:2::ea15 k8sPodName=kube-system/clustermesh-apiserver-5777d847d9-j4vh7 subsys=endpoint error="identity sync was cancelled: context canceled" (1 occurrences)
    ⛑️ The following owners are responsible for reliability of the testsuite: 
        - @cilium/endpoint (no-errors-in-logs:pkg/endpoint)
        - @cilium/sig-clustermesh (.github/workflows/tests-clustermesh-upgrade.yaml)
        - @cilium/ci-structure (.github/workflows/tests-clustermesh-upgrade.yaml)
[cilium-test-1] 1 tests failed
Error: Process completed with exit code 1.
```

Sysdump is too large to attach.

--- a/pkg/endpoint/restore.go
+++ b/pkg/endpoint/restore.go
@@ -304,8 +304,11 @@ func (e *Endpoint) restoreIdentity(regenerator *Regenerator) error {
 				Group: initialGlobalIdentitiesControllerGroup,
 				DoFunc: func(ctx context.Context) (err error) {
 					err = e.allocator.WaitForInitialGlobalIdentities(ctx)
-					if err != nil {
+					switch {
+					case err != nil && !errors.Is(err, context.Canceled):
 						e.getLogger().Warn("Failed while waiting for initial global identities", logfields.Error, err)
+						fallthrough
+					case err != nil:
 						return err
 					}
 					close(gotInitialGlobalIdentities)

